lib/restbase.js

```diff
@@ -49,10 +52,40 @@ function RESTBase (options, req) {
             router: options.router,
             statsd: options.metrics
         };
-
     }
+
+    // set the request id, if appropriate
+    this.setRequestId(req);
+
 }
 
+// sets the request id for this instance and adds it to
+// the request header, if defined
+RESTBase.prototype.setRequestId = function(req) {
+
+    if (this._isMain) {
```

Comment:
I don't think this should ever trigger, as we reuse a single restbase instance as the root for all requests. The per-request stuff is done with makeChild.